### ISSUE TITLE
Check array inside an array function

### ISSUE THREAD

**Issue by goktugyil** [goktugyil/EZSwiftExtensions#17]
let array1 = [1,2,3]
let array2 = [1,2]

array1.containsArray(array2) // return true or false

Need this extension for Array.

**Comment by thellimist** [comment]
http://stackoverflow.com/a/25714995/4335588

**Comment by goktugyil** [comment]
extension Array {
    func arrayContainsArray<S: SequenceType where S.Generator.Element: Equatable>(lookFor: S) -> Bool {
        for item: S.Generator.Element in lookFor {
            if self.contains(item) == false { // Cannot convert value of type 'S.Generator.Element' to expected argument type '@noescape _ throws -> Bool'
                return false
            }
        }
        return true
    }
}

couldnt make it work as an extension

**Comment by avismarahl** [comment]
After a little bit of playing around, I got it right. Here is the code:
```swift
extension Array where Element : Equatable {
    func arrayContainsArray(lookFor:[Element]) -> Bool {
        for item in lookFor {
            if self.contains(item) == false {
                    return false
            }
        }
        return true
    }
}
```

**Comment by avismarahl** [comment]
The reason for previous issue is that when you referenced `self.contains`, the `self`'s `Element` type may not be equatable. There is a type mismatch at `self.contains(item)`. You could solve this by constraining the method to arrays whose elements are equatable.

**Comment by goktugyil** [comment]
Fix issue #17 is not working when I use 

    let arr = [1,2,3,4]
    let arr2 = [1,2,3]

I'm just going to comment it out inside the code.

**Comment by avismarahl** [comment]
I think it does. 
<img width="1256" alt="screen shot 2015-12-28 at 12 12 44 pm" src="https://cloud.githubusercontent.com/assets/5190296/12015064/bf8deb58-ad5c-11e5-82c4-7d6b436cf4b7.png">
